runtime/lua/lsp/client.lua

```diff
@@ -157,16 +157,23 @@ client.request_async = function(self, method, params, cb)
     return nil
   end
 
+  local callback_list
   if cb == nil then
-    cb = get_callback_function(method)
+    callback_list = get_callback_function(method)
+  elseif type(cb) == 'table' then
```

Comment:
callbacks sometimes being a list is a bit messy, better then to always make it a list.

Reply:
They'll always be a list after this call. get_callback_function should changed to get_list_of_callback_functions.

Just wanted people to be able to specify a callback with a function reference directly, rather than in a table

Reply:
> Just wanted people to be able to specify a callback with a function reference directly, rather than in a table.

This doesn't contradict what I said. `add_callback` can represent it as a list either way. At no point do we need to store the union between a callback and a list of callbacks internally.

Reply:
but if someone wants to directly call a request like:

`client:request_async('textDocument/example', { <params> }, callback_function)`

Then we have to accept a function reference directly, rather making them put it in a table

Reply:
(I also noticed that I don't pass callback_list correctly yet, so it's still WIP :) )

Reply:
Sorry, it seems I placed this comment on silghtly the wrong line, indeed `request_async` should  accept a single callback, but then _only_ a single callback. The point is than any value should either be a single callback or a list of callbacks, never the union of the two.